### PR TITLE
Adding s2i tests

### PR DESCRIPTION
**Pull request by VeereshAradhya**
Adding S2I tests which are present in https://github.com/ppitonak/cluster-tasks-tests

### CODE REVIEW

File: testdata/v1beta1/clustertask/pipelines/s2i-dotnet.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: s2i-dotnet-pipeline
+spec:
+  params:
+  - name: EXAMPLE_REVISION
+    description: git branch of .NET example
+    default: dotnetcore-3.1
```
Comment (ppitonak) [inline testdata/v1beta1/clustertask/pipelines/s2i-dotnet.yaml:9]: This cannot be hard-coded, you need to choose correct branch for each version of .Net

**Review by ppitonak — CHANGES_REQUESTED** [review]

File: steps/cli/tkn.go [hunk 2]
```diff
@@ -21,4 +24,30 @@ var _ = gauge.Step("Start and verify pipeline <pipelineName> with param <paramNa
...
+var _ = gauge.Step("Start and verify dotnet pipeline <pipelineName> with values stored in variable <variableName> with workspace <workspaceValue>", func(pipelineName, variableName, workspaceValue string) {
+	values := store.GetScenarioDataSlice(variableName)
+	params := make(map[string]string)
+	workspaces := make(map[string]string)
+	workspaces[strings.Split(workspaceValue, ",")[0]] = strings.Split(workspaceValue, ",")[1]
+	paramName := "VERSION"
+	re := regexp.MustCompile(`\d+\.\d+`)
+	var exampleRevision string
+	for _, value := range values {
+		if value == "latest" {
```
Comment (ppitonak) [inline steps/cli/tkn.go:38]: Shouldn't we add the same also to previous Gauge step, line 22 or so.

File: steps/cli/tkn.go [hunk 3]
```diff
@@ -21,4 +24,30 @@ var _ = gauge.Step("Start and verify pipeline <pipelineName> with param <paramNa
...
+			continue
+		}
+		exampleRevision = re.FindString(value)
+		params[paramName] = value
+		versionInt, _ := strconv.ParseFloat(exampleRevision, 64)
```
Comment (ppitonak) [inline steps/cli/tkn.go:43]: Will parsing as float work for any value?

Comment (VeereshAradhya) [reply steps/cli/tkn.go:43]: You mean, for any number?

File: steps/cli/tkn.go (hunk 2 above)
Comment (VeereshAradhya) [reply steps/cli/tkn.go:41]: Done

File: testdata/v1beta1/clustertask/pipelines/s2i-dotnet.yaml (hunk 1 above)
Comment (VeereshAradhya) [reply testdata/v1beta1/clustertask/pipelines/s2i-dotnet.yaml:9]: This is just the default value, in the step I am passing this value as well

Comment (ppitonak) [reply testdata/v1beta1/clustertask/pipelines/s2i-dotnet.yaml:9]: oh, my bad

File: steps/cli/tkn.go (hunk 3 above)
Comment (ppitonak) [reply steps/cli/tkn.go:46]: OK, looks safe

**Review by ppitonak — APPROVED** [review]

**Pull request merged**